File: atv_academia/Professor.java
import java.util.ArrayList;
import java.util.HashSet;

enum TipoProfessor {
    MUSCULAÇÃO,
    NATAÇÃO,
}

public class Professor extends Funcionario {
    private TipoProfessor tipo;
    private HashSet<Aluno> alunos;

    public Professor(String matricula, String nome, TipoProfessor tipo){
        super(matricula, nome);
        this.tipo = tipo;
        this.alunos = new HashSet<Aluno>();
    }

    public void addAluno(Aluno aluno) {
        this.alunos.add(aluno);
    }

    public void removeAluno(Aluno aluno) {
        this.alunos.remove(aluno);
    }

    public HashSet<Aluno> getAlunos() {
        return alunos;
    }

    public void setTipo(TipoProfessor tipo) {
        this.tipo = tipo;
    }

    public TipoProfessor getTipo() {
        return tipo;
    }
}

File: atv_academia_kotlin/src/main/kotlin/Professor.kt
enum class TipoProfessor {
    MUSCULAÇÃO,
    NATAÇÃO
}

class Professor(matricula: String?, nome: String?, var tipo: TipoProfessor) : Funcionario(
    matricula,
    nome
) {
    private val alunos: HashSet<Aluno>

    init {
        alunos = java.util.HashSet<Aluno>()
    }

    override val tipoFuncionario: String
        get() = String.format("imd.academia.Professor [%s]", tipo.toString())

    fun addAluno(aluno: Aluno) {
        alunos.add(aluno)
    }

    fun removeAluno(aluno: Aluno) {
        alunos.remove(aluno)
    }

    fun getAlunos(): HashSet<Aluno> {
        return alunos
    }
}
